enums.rb

```diff
@@ -0,0 +1,221 @@
+# rubocop:disable Style/For
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/MethodLength
+module Enumerable
+  def my_each
+    if block_given?
+      for element in self
+        yield(element)
+      end
+    else
+      to_enum
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      for i in (0...size)
+        yield(i, self[i])
+      end
+    else
+      to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+
+      arr = []
+      my_each do |element|
+        arr << element if yield(element)
+      end
+      arr
+    else
+      to_enum
+    end
+  end
+
+  def my_all?(arg = nil)
+    response = true
+    if block_given?
+      my_each do |element|
+        response = false unless yield(element)
+      end
+
+    elsif !arg.nil?
+      if arg.is_a? Class
+        my_each do |element|
+          response = false unless element.is_a?(arg)
+        end
+      elsif arg.is_a? Regexp
+        my_each do |element|
+          response = false unless element.match(arg)
+        end
+
+      else
+        my_each do |element|
+          response = false unless element.eql?(arg)
+        end
+      end
+
+    else
+      response = true
+    end
+    response
+  end
+
+  def my_any?(arg = nil)
+    response = false
+    if block_given?
+      my_each do |element|
+        response = true if yield(element)
+      end
```

Comment:
* [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[nil, false, nil, false].my_any? != [nil, false, nil, false].any?
```

Reply:
Fixed this as well :) 
![image](https://user-images.githubusercontent.com/53732843/81596110-c1846e00-9380-11ea-9f42-7d9fdb4d2f89.png)